src/mlpack/methods/reinforcement_learning/q_networks/vanilla_dqn.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file vanilla_dqn.hpp
+ * @author Nishant Kumar
+ *
+ * This file contains the implementation of the vanilla deep q network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_VANILLA_DQN_HPP
+#define MLPACK_METHODS_RL_VANILLA_DQN_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/gaussian_init.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/loss_functions/mean_squared_error.hpp>
+
+namespace mlpack {
+namespace rl {
+
+using namespace mlpack::ann;
+
+template <typename NetworkType = FFN<MeanSquaredError<>,
+                                    GaussianInitialization>>
+class VanillaDQN
+{
+ public:
+  /**
+   * Default constructor.
+   */
+  VanillaDQN() : network()
+  { /* Nothing to do here. */ }
+
+  /**
+   * Construct an instance of VanillaDQN class.
+   *
+   * @param inputDim number of inputs.
+   * @param hiddenDim1 number of neurons in hiddenlayer-1.
+   * @param hiddenDim2 number of neurons in hiddenlayer-2.
+   * @param outputDim number of neurons in output layer.
```

Comment:
```suggestion
   * @param inputDim number of inputs.
   * @param hiddenDim1 Number of neurons in hiddenlayer-1.
   * @param hiddenDim2 Number of neurons in hiddenlayer-2.
   * @param outputDim Number of neurons in output layer.
```

Also, what do you think about using `h1, h2` as parameter name?